Tests/PostgresNIOTests/New/Extensions/PSQLFrontendMessageDecoder.swift

```diff
@@ -0,0 +1,176 @@
+@testable import PostgresNIO
+
+struct PSQLFrontendMessageDecoder: NIOSingleStepByteToMessageDecoder {
+    typealias InboundOut = PSQLFrontendMessage
+    
+    private(set) var isInStartup: Bool
+    
+    init() {
+        self.isInStartup = true
+    }
+    
+    mutating func decode(buffer: inout ByteBuffer) throws -> PSQLFrontendMessage? {
+        // make sure we have at least one byte to read
+        guard buffer.readableBytes > 0 else {
+            return nil
+        }
+        
+        if self.isInStartup {
+            guard buffer.readableBytes >= 4 else {
```

Comment:
This and the other guards on readableBytes are redundant; `getInteger(at:as:)` already does this:

```swift
        guard let range = self.rangeWithinReadableBytes(index: index, length: MemoryLayout<T>.size) else {
            return nil
        }
```
So for that read you can just check that result for `nil` instead of having a separate check. All other `read*()` and `get*()` methods also do this, so you can simplify _and_ get rid of the `!`s.

This all being said, I very, very vaguely once recall there possibly having been a reason _not_ to do that... I have no idea what it was or if it was a real thing and not my fevered imagining, but if so, please tell me 😰